cmake/googletest/CMakeLists.txt

```diff
@@ -17,7 +17,7 @@
 cmake_minimum_required(VERSION 3.16)
 project(googletest-build CXX)
 
-if(BUILD_TEST)
+if(BUILD_TESTING)
```

Comment:
For me this is a API change of the build system. Shouldn't we mention this also in the changelog?

But for me the naming is wrong - sorry for opening this discussion. 

What is `BUILD_TESTING` doing, it builds module and integration tests therefore it should be called `BUILD_TESTS` or `BUILD_TEST` but I prefer the plural here.

I think the confusion may come from the hoofs testing and posh testing package which are building blocks to write tests in iceoryx and are build together with tests....

Reply:
Another thing - but it does not have to be done in this PR, but since you are already touching the right files.
We call it `BUILD_ALL`, `BUILD_DOC` and `BUILD_TEST` but this is inconsistent to `EXAMPLES`, `INTROSPECTION`, and the remaining features which can be build.

Could we add `BUILD_` to all those cmake options which build additional features. The best thing is to start them with a verb and then mention what you are doing which we already do 50% of the times.

So it would be `BUILD_INTROSPECTION`, `USE_CCACHE`, `ADD_COVERAGE_FLAGS`, `SUPPORT_ONE_TO_MANY_CONNECTIONS`, `ADD_SANITIZE_FLAGS` and so on ... Then it would be more consistent and more expressive

Reply:
Regarding `BUILD_TESTING`, this is somewhat the convention with cmake. At least when CTest is used and one would expect that tests can be switched on/off with this flag.

Regarding the general renaming. I think it is a good idea but should be done in a separate PR since this already has it's own issues with the ROS CI